test/converters/converters.go

```diff
@@ -0,0 +1,21 @@
+package converters
+
+import (
+	"github.com/bitrise-io/steps-deploy-to-bitrise-io/test/converters/junitxml"
+	"github.com/bitrise-io/steps-deploy-to-bitrise-io/test/converters/xcresult"
+)
+
+type Intf interface {
+	SetFiles([]string)
+	XML() ([]byte, error)
```

Comment:
i would just convert to a shared test results model and return the model, that model could be parsed on the client side (as in case of every converter we are looking for the same XML format, described by the same model).
this would also lead to a better name, for example: `Convert/Parse...`
this way also the interface could be: `TestResultConverter`

Reply:
I would keep the XML output as each platform has different way to produce an xml. Each platform has a different logic to build up the Junit XML structure.

Reply:
So I think this is why it needs to be handled individually.